src/Ordered.java

```diff
@@ -68,29 +66,32 @@ public boolean decreaseQuantity(){
         }
     }
 
-    public void setEstimatedTotal () {
-        try{
+    private BigDecimal getEstimatedTotal () throws QuantatyException{
         if (this.decreaseQuantity()) {
-            this.estimatedTotal = this.getProduct().getPrice().multiply(new BigDecimal(this.getQuantity()));
+            this.estimatedTotal = (this.getProduct().getPrice().multiply(BigDecimal.valueOf(this.getQuantity())));
             this.paymentMethod.setValue(this.estimatedTotal);
+            return this.estimatedTotal;
         }
         else {
-            System.out.println("quantidade de produto insuficiênte");
-        }
-        }
-        catch(NullPointerException exception){
-            System.out.println("O preço do produto não está cadastrado");
+            throw new QuantatyException("A quantidade de produto é insuficiênte para esta compra!");
         }
     }
 
-    public void discountEstimatedTotal (BigDecimal percentDiscount) {
-        try{
-        BigDecimal subtotalDiscount = (getEstimatedTotal().subtract((getEstimatedTotal().multiply(percentDiscount)).divide(new BigDecimal(100))));
-        System.out.println("O novo valor do produto com o desconto de " + percentDiscount + "% concedido é: R$" + subtotalDiscount);
-        }
-        catch(NullPointerException exception){
-            System.out.println("O preço do produto não está cadastrado");
-        }
+    public BigDecimal getEstimatedTotalValue (){
+        return this.estimatedTotal;
     }
 
+    public void setEstimatedTotal(BigDecimal estimatedTotal) {
+        this.estimatedTotal = estimatedTotal;
+        this.paymentMethod.setValue(this.estimatedTotal);
+    }
+
+    public void setSubtotalDiscount (Integer percentDiscount) {
+        this.subtotalDiscount = (getEstimatedTotalValue().subtract((getEstimatedTotalValue().multiply(BigDecimal.valueOf(percentDiscount))).divide(BigDecimal.valueOf(100))));
```

Comment:
Esta linha possui 2 chamadas para `getEstimatedTotalValue()` que por sua vez chama dentro de um if `this.decreaseQuantity()`. Isso não fará com que a quantidade seja diminuida 2 vezes em cada chamada?

Reply:
Antes estava acontecendo isso, porque ai nessa linha eu chamava o getEstimatedTotal (esse faz a conta), mas agora eu chamo o get EstimatedTotalValue (que só pega o valor).